### ISSUE TITLE
cargo-binstall 0.21.2 sometimes (?) failed to install cargo-watch

### ISSUE THREAD

**Issue by taiki-e** [taiki-e/install-action#89]
I noticed that one of the installations using binstall fallback failed in CI on the latest main branch.
I'm not sure if it is possible to reproduce this deterministically since other jobs have been successful, but there seems to be some kind of regression.

https://github.com/taiki-e/install-action/actions/runs/4411680156/jobs/7730426642

```
 INFO resolve: Resolving package: 'cargo-watch'
 INFO resolve: Resolving package: 'watchexec-cli'
ERROR Fatal error:

  × For crate cargo-watch: Failed to download from remote: could not
  │ GET https://crates.io/api/v1/crates/cargo-watch/8.4.0/download: HTTP
  │ status server error (500 Internal Server Error) for url (https://
  │ static.crates.io/crates/cargo-watch/cargo-watch-8.4.0.crate)
  ├─▶ Failed to download from remote: could not GET https://crates.io/api/
  │   v1/crates/cargo-watch/8.4.0/download: HTTP status server error (500
  │   Internal Server Error) for url (https://static.crates.io/crates/cargo-
  │   watch/cargo-watch-8.4.0.crate)
  ├─▶ could not GET https://crates.io/api/v1/crates/cargo-watch/8.4.0/
  │   download: HTTP status server error (500 Internal Server Error) for url
  │   (https://static.crates.io/crates/cargo-watch/cargo-watch-8.4.0.crate)
  ╰─▶ HTTP status server error (500 Internal Server Error) for url (https://
      static.crates.io/crates/cargo-watch/cargo-watch-8.4.0.crate)
```

cc @NobodyXu

**Comment by NobodyXu** [comment]
Thanks!
In v0.21.x, we replace crates_io_api with directly fetching from https://crates.io and use the built-in rate limit of binstalk-downloader

This error reminded me of quickinstall limits requests to https://crates.io to one per second, and that seems to be the right thing to do, since the [crates.io crawler policy](https://crates.io/policies#crawlers) says the same thing:

> We allow our API and website to be crawled by commercial crawlers such as GoogleBot. At our discretion, we may choose to allow access to experimental crawlers, as long as they limit their request rate to 1 request per second or less.

I will create a PR for this and make another patch release to fix this.

**Comment by NobodyXu** [comment]
@taiki-e I've released cargo-binstall v0.21.3 which should fix this issue.
